### PR TITLE
add replace argument name method

### PR DESCRIPTION
**Pull request by nllong**
* The `get_span` method is causing some issues. There seems to be many conditions on the `get_span` method. @macintoshpie -- can you take a look at the one failing test. There is a extra `1` in the ChildB element which is from the original list. This is caused by the `get_span` method. Let me know what you think!

### CODE REVIEW

File: modelica_builder/edit.py [hunk 1]
```diff
@@ -92,8 +92,13 @@ def apply_edits(cls, edits, document):
         # entire document for every edit
         # O(m*n), m = bytes in document, n = number of edits
         for edit in ordered_edits:
-            # remove edit span
-            if edit.start != edit.stop:
+            # catch the case where the name is to replaced which is only one character. The `get_span` method
+            # will set the edit.stop to None if the length is only one.
+            if edit.stop is None:
```
Comment (nllong) [inline modelica_builder/edit.py:97]: This section should really be handled in the `get_span` method.

File: modelica_builder/model.py [hunk 2]
```diff
@@ -18,6 +17,7 @@
 from modelica_builder.edit import Edit
 from modelica_builder.selector import (
     ComponentDeclarationSelector,
+    ComponentModificationNameSelector,
```
Comment (nllong) [inline modelica_builder/model.py:20]: This is the new method

File: modelica_builder/transformation.py [hunk 3]
```diff
@@ -105,7 +105,7 @@ def make_edits_for_modifications(class_modification_node, modifications, parser,
                     depth=depth + 1,
                     indented=indented)
             else:
-                edit = Edit.make_replace(f'={requested_modification_value}')
+                edit = Edit.make_replace(f'{requested_modification_value}')
```
Comment (nllong) [inline modelica_builder/transformation.py:108]: the `=` sign shouldn't be in the replace. This should really be part of the `get_span` method when there is a`ModificationContext`.

File: tests/test_model.py [hunk 4]
```diff
@@ -196,7 +195,8 @@ def test_model_insert_component(self):
         self.result = model.execute()
 
         # Assert
-        self.assertHasAdditions(source_file, self.result, ['FancyClass myInstance(arg1=1234) "my comment" annotation(my annotation);'])
+        self.assertHasAdditions(source_file, self.result,
```
Comment (nllong) [inline tests/test_model.py:198]: most of this as a result of autopep

File: tests/test_model.py [hunk 5]
```diff
@@ -516,6 +522,23 @@ def test_model_update_component_modifications_simple(self):
             'Resistor R(R=100);'
         ])
 
+    def test_model_rename_argument(self):
+        source_file = os.path.join(self.data_dir, 'DCMotor.mo')
```
Comment (nllong) [inline tests/test_model.py:526]: very simple test!

File: modelica_builder/transformation.py (hunk 3 above)
Comment (macintoshpie) [reply modelica_builder/transformation.py]: I think we do want this here, the node we're modifying, `modification`, where we _aren't_ doing a class modification (the else case we are in here) is defined in the grammar like this:
```
modification
   : class_modification ('=' expression)?
   | '=' expression
   | ':=' expression
   ;
```
ie we are overwriting the `=` and the `expression`

File: modelica_builder/edit.py (hunk 1 above)
Comment (macintoshpie) [reply modelica_builder/edit.py]: I just added a fix for this in get_span - it now returns a range which is more pythonic, where the stop == index of end of node text + 1.

**Review by macintoshpie — APPROVED** [review]
This looks good! Can you take a look at my changes? I tried to simplify the fix to our problem (which I describe below)

You were correct that `get_span` returning a span for a single character was causing problems down the line. This is because we use `Edit`'s start and stop to indicate a range to delete (and then insert at start). If start and stop are the same (which it was for single char replacements) then it wouldn't actually replace it, it would just insert.

The fix is to have get_span return start, stop + 1 (like how python uses slice indices). I just had to update two other places that use this (insert factory where after=True as well as the apply_edits which no longer needs a +1 in a range).

The result is we get to keep our existing general get_span and apply_edits.

File: modelica_builder/edit.py [hunk 6]
```diff
@@ -69,7 +69,7 @@ def make_insert(cls, data, insert_after=True):
         def insert(node):
             start, stop = modelica_parser.get_span(node)
             if insert_after:
-                start = stop + 1
+                start = stop
```
Comment (nllong) [inline modelica_builder/edit.py:72]: nice catch here @macintoshpie

File: modelica_builder/edit.py [hunk 7]
```diff
@@ -92,14 +92,9 @@ def apply_edits(cls, edits, document):
...
-            # catch the case where the name is to replaced which is only one character. The `get_span` method
-            # will set the edit.stop to None if the length is only one.
-            if edit.stop is None:
-                # if the length was 0 then, return the document removing only the one 1 character.
-                document = document[:edit.start] + document[edit.start + 1:]
-            elif edit.start != edit.stop:
-                # remove edit span
-                document = document[:edit.start] + document[edit.stop + 1:]
+            # remove edit span
+            if edit.start != edit.stop:
```
Comment (nllong) [inline modelica_builder/edit.py:96]: yes, this looks better.

File: modelica_builder/edit.py (hunk 6 above)
Comment (macintoshpie) [reply modelica_builder/edit.py:72]: tests FTW

**Review by nllong — COMMENTED** [review]
Thanks for fixing everything. I think this is ready to merge!

**Pull request merged**